pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -442,151 +442,134 @@ contract StablePhantomPool is IRateProvider, BaseGeneralPool, ProtocolFeeCache {
         uint256 indexIn,
         uint256 indexOut,
         uint256[] memory scalingFactors
-    ) private returns (uint256) {
+    ) private returns (uint256 result) {
         _upscaleArray(balances, scalingFactors);
 
         (uint256 virtualSupply, uint256[] memory balancesWithoutBpt) = _payProtocolFeesBeforeJoinExit(balances);
 
-        bool bptIsTokenIn = swapRequest.tokenIn == IERC20(this);
+        bool isJoinSwap = swapRequest.tokenOut == IERC20(this);
+        bool isGivenIn = swapRequest.kind == IVault.SwapKind.GIVEN_IN;
+
+        uint256 upscaledAmount = _upscale(swapRequest.amount, scalingFactors[isGivenIn ? indexIn : indexOut]);
         (uint256 amp, ) = _getAmplificationParameter();
 
         // The lower level function return values are still upscaled, so we need to downscale the final return value
-        if (swapRequest.kind == IVault.SwapKind.GIVEN_IN) {
-            // Returning amountOut; tokens are leaving the Vault, so we round down
-            return
-                _downscaleDown(
-                    _onSwapBptGivenIn(
-                        _upscale(swapRequest.amount, scalingFactors[indexIn]),
-                        indexIn,
-                        indexOut,
-                        bptIsTokenIn,
-                        amp,
-                        virtualSupply,
-                        balancesWithoutBpt
-                    ),
-                    scalingFactors[indexOut]
-                );
+        if (isJoinSwap) {
+            uint256 indexInNoBpt = _skipBptIndex(indexIn);
+            uint256 unknownAmount = _onSwapBptJoin(
```

Comment:
```suggestion
            uint256 calculatedAmount = _onSwapBptJoin(
```
The original name sounds better to me. It's "known" when it's returned - calculated - so it seems odd to name the return value "unknown." I believe we've used "calculated" elsewhere and in the docs (e.g., amountCalculated in the Vault swaps contract).

Reply:
Yes, this is Vault terminology. For example, swaps deal with `amountGiven` and `amountCalculated`.